specs/newton/approved/initial_design.rst

```diff
@@ -0,0 +1,61 @@
+..
+ This work is licensed under a Creative Commons Attribution 3.0 Unported
+ License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+
+========================
+Initial design of Ephyra
+========================
+
+Ephyra is a simple workload migrator for OpenStack.
+
+Problem description
+===================
+
+Imagine that you have a pool of Ironic nodes running a Hadoop workload. You would like to do Continuous Deployment of the control plane for this OpenStack deployment, while having a graceful rollback procedure for when the new version of the control plane is broken for some reason. Specifically for this thought experiment, you can assume that only Ironic is supported as a Nova hypervisor in this deployment, and that the workload running on those Ironic nodes can handle partial outages without degrading performance.
+
+The solution we have come to is that you have two OpenStack deployments. In the initial state, the first of these deployments (which we shall call CP1) has all of the Ironic nodes registered with it, and the second deployment (CP2) has none. The Continous Deployment system detects that there is a new version of OpenStack to deploy. It installs that version onto CP2 in whatever manner it desires. Installation of OpenStack is outside the scope of Ephyra, but it is assumed to be automated.
```

Comment:
I think it needs to be clear that Ephyra is not in any way involved in the rollout of the CP openstack Deploys.